src/Compat/FormDecorator/PrimaryButtonDecorator.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace ipl\Web\Compat\FormDecorator;
+
+use ipl\Html\Contract\DecorationResult;
+use ipl\Html\Contract\Form;
+use ipl\Html\Contract\FormDecoration;
+use ipl\Html\Contract\FormSubmitElement;
+
+class PrimaryButtonDecorator implements FormDecoration
+{
+    public function decorateForm(DecorationResult $result, Form $form): void
+    {
+        $primaryButtonCandidate = null;
+        foreach ($form->getElements() as $element) {
+            if (! $element instanceof FormSubmitElement) {
+                continue;
+            }
+
+            $classes = implode(' ', (array) $element->getAttributes()->get('class')->getValue());
+
+            if (preg_match('/(?<=^|\s)btn-primary(?=$|\s)/', $classes)) {
+                return;
+            }
+
+            if (! $primaryButtonCandidate && ! preg_match('/\bbtn-[\w-]+/', $classes)) {
+                $primaryButtonCandidate = $element;
+            }
```

Comment:
Can't you just cast the value to an array and handle it as such? I think it's bad practice to define multiple classes as one single string anyway, exactly due to this reason. It would make searching for other `btn-*` classes also easier.

Have you found cases where button classes are defined in a string with other classes or are you only anticipating such?